pkg/crc/services/dns/dns_darwin.go

```diff
@@ -37,7 +39,8 @@ func runPostStartForOS(serviceConfig services.ServicePostStartConfig, result *se
 	}
 
 	// Write resolver config to host
-	needRestart, err := createResolverFile(serviceConfig.IP, filepath.Join("/", "etc", "resolver", serviceConfig.BundleMetadata.ClusterInfo.BaseDomain))
+	needRestart, err := createResolverFile(serviceConfig.IP, serviceConfig.BundleMetadata.ClusterInfo.BaseDomain,
+		filepath.Join("/", "etc", "resolver", serviceConfig.BundleMetadata.ClusterInfo.BaseDomain))
```

Comment:
I would move that `filepath.Join()` to `createResolverFile`, as we are not going to be creating resolver files outside of `/etc/resolver/` anyway.

Reply:
`createResolverFile` This function is independent of file so I don't want to move that to this functio, to me provide the info and file path to this function and it will create the content for a resolve file, it happened to be a static file in our case if we really want to do then might be change the function name to `createBaseDomainResolverFile` or something.

Reply:
oh, forgot to suggest to rename the file from `$domain` to `crc-resolver` or something. 

> This function is independent of file 
And my point is that it does not make sense to expect the caller to know that `createResolverFile` must be used with a file based in `/etc/resolver/` :) we can keep a `filename` arg if you think it's useful, but I'd still move the `filepath.Join("/", "etc", "resolver", filename)` call in `createResolverFile`.